pkg/common/config/config.go

```diff
@@ -255,23 +256,41 @@ func (cfg *Config) FromEnv() error {
 // ReadConfig parses vSphere cloud config file and stores it into VSphereConfig.
 // Environment variables are also checked
 func ReadConfig(byConfig []byte) (*Config, error) {
+	var cfg *Config
+
 	if len(byConfig) == 0 {
 		return nil, fmt.Errorf("Invalid YAML/INI file")
 	}
 
-	cfg, err := ReadConfigYAML(byConfig)
-	if err != nil {
-		klog.Warningf("ReadConfigYAML failed: %s", err)
+	// Check to see if yaml file before calling parser to prevent unnecessary error messages.
+	yamlErr := isConfigYaml(byConfig)
+	if yamlErr == nil {
+		cfg, yamlErr = ReadConfigYAML(byConfig)
 
-		cfg, err = ReadConfigINI(byConfig)
-		if err != nil {
-			klog.Errorf("ReadConfigINI failed: %s", err)
-			return nil, err
+		// ReadConfigYAML can fail for other config errors not related to YAML.  So it is ok for yamlErr == nil and a
+		// new error occurs
+		if yamlErr != nil {
+			klog.Errorf("ReadConfigYAML failed: %s", yamlErr)
+			return nil, yamlErr
 		}
+		klog.Info("ReadConfig YAML succeeded")
+	} else {
+		klog.Infof("YAML not detected.  Attempting INI config load.")
```

Comment:
`klog.Info` is pretty high level for this. This should be may be - `klog.V(4)`

Reply:
I am ok with changing the level.  Let me make those changes.